orttraining/orttraining/core/framework/checkpointing.cc

```diff
@@ -125,48 +52,77 @@ std::vector<std::string> GetOrderedOrtValueNames(const NameMLValMap& name_to_val
   return ordered_names;
 }
 
+}  // namespace
+
 Status SaveRuntimeTensors(
     const PathString& tensors_path,
     const PathString& tensors_data_path,
     const DataTransferManager& data_transfer_manager,
-    const NameMLValMap& ort_values) {
+    const NameMLValMap& ort_values,
+    bool force_save_as_external_data) {
   // just write data file basename to TensorProto - this will get overwritten
   //   with the actual path when loading the checkpoint
   const PathString tensors_data_relative_path = GetLastComponent(tensors_data_path);
-
   const std::vector<std::string> ordered_tensor_names = GetOrderedOrtValueNames(ort_values);
+  bool save_as_external_data = force_save_as_external_data;
+
+  // Check whether there is potential protobuf size limit was hit.
+  if (!save_as_external_data) {
```

Comment:
There must be a simpler way to do this check...

Reply:
Size of the tensor is used on line 104 as well... can we just do this check there? We can simply store data as external only for the tensor which crosses the 2GB limit.

Reply:
revert the change in checkpointing.h/cc as talked offline.

simplify the size check in checkpoint.cc, we accumulate the size until hit the up limit.